presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -613,8 +633,6 @@ public ListeningExecutorService createResourceManagerExecutor(ResourceManagerCon
         // presto announcement
         checkArgument(!(serverConfig.isResourceManager() && serverConfig.isCoordinator()),
                 "Server cannot be configured as both resource manager and coordinator");
-        checkArgument(!(serverConfig.isResourceManager() && serverConfig.isCatalogServer()),
-                "Server cannot be configured as both resource manager and catalog server");
```

Comment:
Why are we removing this check?
If we don't need it, we shouldn't add it in the first commit

Reply:
Commits will be squashed so this should be gone